packages/frontend/src/pages/Dashboard.tsx

```diff
@@ -400,28 +400,37 @@ const Dashboard: FC = () => {
     }, [haveTilesChanged, haveFiltersChanged, isEditMode]);
 
     useEffect(() => {
-        const createChartInDashboardFlow = sessionStorage.getItem(
-            'unsavedDashboardTiles',
-        );
-        history.block((prompt) => {
-            if (
-                isEditMode &&
-                (haveTilesChanged || haveFiltersChanged) &&
-                !prompt.pathname.includes(
-                    `/projects/${projectUuid}/dashboards/${dashboardUuid}`,
-                ) &&
-                createChartInDashboardFlow
-            ) {
-                setBlockedNavigationLocation(prompt.pathname);
-                setIsSaveWarningModalOpen(true);
-                return false; // blocks history
-            }
-            return undefined; // allow history
-        });
+        // Check if in edit mode and changes have been made
+        if (isEditMode && (haveTilesChanged || haveFiltersChanged)) {
+            // Define the navigation block function
+            const navigationBlockFunction = (prompt: { pathname: string }) => {
+                // Check if the user is navigating away from the current dashboard
+                if (
+                    !prompt.pathname.includes(
+                        `/projects/${projectUuid}/dashboards/${dashboardUuid}`,
+                    ) &&
+                    // Allow user to add a new table
+                    !sessionStorage.getItem('unsavedDashboardTiles')
```

Comment:
Checks if user is adding a new chart -> then dont show a warning

Reply:
Cool. Just so you know (if you are looking in the future) I'm probably going to change where these go in the near future, but this is fine for now.